file.cc

```diff
@@ -51,179 +33,92 @@ bool tryPossibleZipName( std::string const & name, std::string & copyTo )
     return false;
 }
 
-void loadFromFile( std::string const & n, std::vector< char > & data )
+void loadFromFile( std::string const & filename, std::vector< char > & data )
 {
-  File::Class f( n, "rb" );
-
-  f.seekEnd();
-
-  data.resize( f.tell() );
-
-  f.rewind();
-
-  f.read( &data.front(), data.size() );
+  File::Class f( filename, "rb" );
+  QByteArray byteArray{f.readall()};
+  data                 = std::vector< char >( byteArray.begin(), byteArray.end() );
 }
 
 bool exists( char const * filename ) noexcept
 {
-#ifdef __WIN32
-#if defined(__WIN64) || defined(_MSC_VER)
-  struct _stat64 buf;
-#else
-  struct _stat buf;
-#endif
-  wchar_t wname[16384];
-  MultiByteToWideChar( CP_UTF8, 0, filename, -1, wname, 16384 );
-#if defined(__WIN64) || defined(_MSC_VER)
-  return _wstat64( wname, &buf ) == 0;
-#else
-  return _wstat( wname, &buf ) == 0;
-#endif
-#else
-  struct stat buf;
-
-  // EOVERFLOW rationale: if the file is too large, it still does exist
-  return stat( filename, &buf ) == 0 || errno == EOVERFLOW;
-#endif
+  return QFileInfo( QString::fromStdString( filename ) ).exists();
 }
 
-void Class::open( char const * filename, char const * mode ) 
+void Class::open( char const * mode )
 {
   QFile::OpenMode openMode = QIODevice::Text;
-  const char * pch = mode;
-  while( *pch )
-  {
-    switch( *pch )
-    {
-      case 'r': openMode |= QIODevice::ReadOnly;
-                break;
-      case 'w': openMode |= QIODevice::WriteOnly;
-                break;
-      case '+': openMode &= ~( QIODevice::ReadOnly | QIODevice::WriteOnly );
-                openMode |= QIODevice::ReadWrite;
-                break;
-      case 'a': openMode |= QIODevice::Append;
-                break;
-      case 'b': openMode &= ~QIODevice::Text;
-                break;
-      default:  break;
+
+  for ( const char * i = mode; *i != '\0'; ++i ) {
+    switch ( *i ) {
+      case 'r':
+        openMode |= QIODevice::ReadOnly;
+        break;
+      case 'w':
+        openMode |= QIODevice::WriteOnly;
+        break;
+      case '+':
+        openMode &= ~( QIODevice::ReadOnly | QIODevice::WriteOnly );
+        openMode |= QIODevice::ReadWrite;
+        break;
+      case 'a':
+        openMode |= QIODevice::Append;
+        break;
+      case 'b':
+        openMode &= ~QIODevice::Text;
+        break;
+      default:
+        break;
     }
-    ++pch;
   }
 
-  f.setFileName( filename );
 
   if ( !f.open( openMode ) )
-    throw exCantOpen( std::string( filename ) + ": " + f.errorString().toUtf8().data() );
+    throw exCantOpen( f.fileName().toStdString() + ": " + f.errorString().toUtf8().data() );
 }
```

Comment:
When f.fileName() was setted a value

Reply:
At the constructor of File::Class. The name is guaranteed to exist
![image](https://user-images.githubusercontent.com/20123683/232037827-ffd269ed-44e4-4820-bbb3-3d5d3b1e9b06.png)

Reply:
So open(mode) should be a private method only used internally.

Will there be some cases when init `File::Class a(filename1)` and then open another file `a.open(filename2,mode)`?